### PR TITLE
Add strict jsx logical expressions eslint rule

### PR DESCRIPTION
**Pull request by jolle**
#### Summary

Adds the `jsx-expressions/strict-logical-expressions` eslint rule to catch potential incorrect behaviour with logical expressions in JSX like `{0 && …}` and `{"" && …}`. Also fixes existing offenders of this rule in the code.

### CODE REVIEW

File: frontend/src/citizen-frontend/map/UnitDetailsPanel.tsx [hunk 1]
```diff
@@ -166,7 +166,7 @@ export default React.memo(function UnitDetailsPanel({
             <div>
               {unit.roundTheClock ? t.map.shiftCareYes : t.map.shiftCareNo}
             </div>
-            {unit.url && (
+            {unit.url !== null && (
```
Comment (jaahon) [inline frontend/src/citizen-frontend/map/UnitDetailsPanel.tsx:169]: I think the correct fix in a lot of cases (such as here) is to convert these into `!!unit.url` to keep functionality the same. Like here, I don't think it makes sense to have a homepage link that points to nowhere.

File: frontend/src/employee-frontend/components/common/Filters.tsx [hunk 2]
```diff
@@ -112,7 +112,7 @@ export function Filters({
   const { i18n } = useTranslation()
   return (
     <FiltersContainer>
-      {setFreeText && searchPlaceholder && (
+      {setFreeText && !!searchPlaceholder && (
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/common/Filters.tsx:115]: Not related to your changes, but this condition just seems wrong. Hiding the whole input if a placeholder is missing or empty 😄 

Could you fix this as well?

Comment (jolle) [reply frontend/src/employee-frontend/components/common/Filters.tsx]: Sure! 😄

**Review by jaahon — APPROVED** [review]
Good job!

**Pull request merged**